src/components/screens/manage-visitor-session-variables-screen.tsx

```diff
@@ -0,0 +1,732 @@
+        );
+
+        if (newVisitors.length > 0) {
+          newVisitors.forEach(visitor => visitorIds.push(visitor));
+        }
+      });
+
+      const visitorsApi = Api.getVisitorsApi(accessToken);
+      const foundVisitors = await Promise.all(
+        visitorIds.map(visitorId => visitorsApi.findVisitor({ exhibitionId, visitorId }))
+      );
+  
+      this.setState({
+        foundSessions,
+        foundVisitors,
+        loading: false
+      });
+    } catch (error) {
+      this.setState({ error });
+    }
+  }
+
+  /**
+   * Event handler for mqtt error
+   *
+   * @param error mqtt error
+   */
+  private onMqttError = (error: Error) => {
+    this.setState({
+      error: error
+    });
+  }
+
+  /**
+   * Sets selected session
+   * 
+   * @param selectedSession selected session
+   */
+  private setSelectedSession = (selectedSession: VisitorSession) => async () => {
+    const { accessToken, exhibitionId } = this.props;
+    if (!accessToken) {
+      return;
+    }
+
+    const sessionVariables = selectedSession.variables;
+    if (sessionVariables && sessionVariables.length > 0) {
+      this.setState({ loading: true });
+
+      const visitorVariablesApi = Api.getVisitorVariablesApi(accessToken);
+      const visitorVariables = await visitorVariablesApi.listVisitorVariables({ exhibitionId });
+      const modifiableVariables = visitorVariables.filter(variable => variable.editableFromUI);
+
+      const variableDataList: SessionVariableData[] = [];
+      for (const sessionVariable of sessionVariables) {
+        const visitorVariable = modifiableVariables.find(variable => variable.name === sessionVariable.name);
+        if (visitorVariable) {
+          variableDataList.push({
+            visitorVariable,
+            sessionVariable
+          });
+        }
+      }
+
+      this.setState({
+        variableDataList,
+        loading: false
+      });
+    }
+
+    this.setState({ selectedSession });
+  }
+
+  /**
+   * Sets selected session variable
+   * 
+   * @param selectedVariableData selected variable data
+   */
+  private setSelectedSessionVariable = (selectedVariableData: SessionVariableData) => () => {
+    this.setState({ selectedVariableData });
+  }
+
+  /**
+   * Event handler for variable value change
+   * 
+   * @param type visitor variable type
+   * @param event React change event
+   */
+  private onVariableValueChange = (type: VisitorVariableType) => (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { selectedSession, selectedVariableData, variableDataList } = this.state;
+    if (!selectedSession || !selectedVariableData || !variableDataList) {
+      return;
+    }
+
+    const updatedValue = this.getUpdatedVariableValue(type, event);
+
+    this.setState(
+      produce((draft: State) => {
+        draft.selectedVariableData!.sessionVariable.value = updatedValue;
+        draft.dataChanged = true;
+        draft.variableDataList = variableDataList!.map(variableData =>
```

Comment:
variableDataList is already null-checked above 
```suggestion
        draft.variableDataList = variableDataList.map(variableData =>
```